src/main/java/com/mraof/minestuck/world/gen/feature/structure/FrogTemplePiece.java

```diff
@@ -0,0 +1,330 @@
+		buildPillars(columnBlock, worldIn, boundingBoxIn, randomIn);
+		buildFrog(stoneBlock, worldIn, boundingBoxIn);
+		
+		return true;
+	}
+	
+	private void generateLoot(IWorld worldIn, MutableBoundingBox boundingBoxIn, Random randomIn, Direction direction, ChunkPos chunkPos)
+	{
+		Debug.debugf("generateLoot");
+		//Direction modifiedDirection = direction.rotateYCCW();
+		//setBlockState(worldIn, Blocks.CHEST.getDefaultState().with(ChestBlock.FACING, this.getCoordBaseMode()), 20 + 20, 17, 11 + 38 + 20, boundingBox);
+		//setBlockState(worldIn, Blocks.CHEST.getDefaultState().with(ChestBlock.FACING, this.getCoordBaseMode()), 19 + 20, 17, 11 + 38 + 20, boundingBox);
+		BlockPos chestPos = new BlockPos(this.getXWithOffset(19 + 20, 11 + 38 + 20), this.getYWithOffset(17), this.getZWithOffset(19 + 20, 11 + 38 + 20));
+		StructureBlockUtil.placeLootChest(chestPos, worldIn, boundingBoxIn, getCoordBaseMode(), ChestType.RIGHT, MSLootTables.FROG_TEMPLE_CHEST, randomIn);
+		chestPos = new BlockPos(this.getXWithOffset(18 + 20, 11 + 38 + 20), this.getYWithOffset(17), this.getZWithOffset(18 + 20, 11 + 38 + 20));
+		StructureBlockUtil.placeLootChest(chestPos, worldIn, boundingBoxIn, getCoordBaseMode(), ChestType.LEFT, MSLootTables.FROG_TEMPLE_CHEST, randomIn);
+		//setBlockState(worldIn, Blocks.CHEST.getDefaultState().with(ChestBlock.FACING, direction), 20 + 20, 17, 11 + 38 + 20, boundingBox);
+		
+		/*
+		//TODO There is only a percent chance of the rotateYCCW applying correctly
+		setBlockState(worldIn, MSBlocks.LOTUS_TIME_CAPSULE_BLOCK.getMainBlock().getDefaultState().with(LotusTimeCapsuleBlock.FACING, Direction.NORTH), 21 + 20, 49, 23 + 38 + 20, boundingBox);
+		//modifiedDirection = modifiedDirection.rotateYCCW();
+		setBlockState(worldIn, MSBlocks.LOTUS_TIME_CAPSULE_BLOCK.getMainBlock().getDefaultState().with(LotusTimeCapsuleBlock.FACING, Direction.EAST), 21 + 20, 49, 22 + 38 + 20, boundingBox);
+		//modifiedDirection = modifiedDirection.rotateYCCW();
+		setBlockState(worldIn, MSBlocks.LOTUS_TIME_CAPSULE_BLOCK.getMainBlock().getDefaultState().with(LotusTimeCapsuleBlock.FACING, Direction.SOUTH), 20 + 20, 49, 22 + 38 + 20, boundingBox);
+		//modifiedDirection = modifiedDirection.rotateYCCW();
+		setBlockState(worldIn, MSBlocks.LOTUS_TIME_CAPSULE_BLOCK.getMainBlock().getDefaultState().with(LotusTimeCapsuleBlock.FACING, Direction.WEST), 20 + 20, 49, 23 + 38 + 20, boundingBox);
+		*/
+		
+		BlockPos lotusBlockPos = new BlockPos(this.getXWithOffset(20 + 20, 22 + 38 + 20), this.getYWithOffset(49), this.getZWithOffset(20 + 20, 22 + 38 + 20)); //TODO temples whose main entrance opens south(and presumedly west) generate one block closer to entrance
+		MSBlocks.LOTUS_TIME_CAPSULE_BLOCK.placeWithRotation(worldIn, lotusBlockPos, getRotation());
+		
+		//TODO Everything seems to generate several times over, first noticed with lotus flower entity, find better fix than below
+		
+		if(!createRan)
```

Comment:
A boolean check similar to another structure piece was added, is that acceptable?

Reply:
Perhaps the confusion was over the gap between the check for createRan and the TODO. createRan is the check

Reply:
The `createRan` check helps making sure that the entity is only placed once. It is not the same thing as making sure that the entity is placed only when the structure should generate for that position.

Reply:
What would I put in order to check what you are talking about?

Reply:
You'd need to calculate the entity position before making the entity, that you could then use to check if the position is within the bounding box with `boundingBoxIn.isVecInside(position)`

Reply:
Added it as an additional check

Reply:
Yeah, that's good. Though as a code style thing, you should now get the entity coordinates for `setLocationAndAngles()` from `entityVec`, so that you're not defining what is meant to be the same coordinates in two different places.

Reply:
And for future reference (not saying that you have to change anything here or now), know that `BlockPos `extends `Vec3i`, meaning that you can use a `BlockPos` in place of an `Vec3i`